firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigFetchHttpClient.java

```diff
@@ -207,9 +230,30 @@ FetchResponse fetch(
     }
 
     ConfigContainer fetchedConfigs = extractConfigs(fetchResponse, currentTime);
+
+    long endTime = System.currentTimeMillis();
+    ClientLogEvent clientLogEvent = generateClientLogEvent(instanceId, startTime, endTime);
+    transport.send(Event.ofTelemetry(clientLogEvent));
+
     return FetchResponse.forBackendUpdatesFetched(fetchedConfigs, fetchResponseETag);
   }
 
+  @VisibleForTesting
+  public ClientLogEvent generateClientLogEvent(String instanceId, long startTime, long endTime) {
```

Comment:
Following on the ideas of differentiating `Transport` config and injecting the `Transport` instance, I think we could encapsulate logging logic in a class and inject an instance of that class, eg
```
class RemoteConfigLogger {
  public RemoteConfigLogger (transport) { ... }

  // A static creation method that initializes a RemoteConfigLogger with a Transport instance might be helpful too

  logFetchEvent(latency){ ... }
}
...
class ConfigFetchHttpClient {
  public ConfigFetchHttpClient ( ... remoteConfigLogger) { ... }
  ...
  FetchResponse fetch(...) {
    ... remoteConfigLogger.logFetchEvent(endTime - startTime)
  }
```

As a side benefit, encapsulating log logic in a class give us one place to look when debugging logging logic, comparable to https://github.com/firebase/firebase-js-sdk/blob/master/packages/remote-config/src/errors.ts

Related, here's a (copy of a) previous project that used the same pattern, and also enabled external devs to subscribe to the same events, which we might consider if folks want to use their own analytics: https://github.com/teju/Example_Code/blob/master/digits-android-master/digits/src/main/java/com/digits/sdk/android/DigitsEventCollector.java

Reply:
I've followed your advice and moved the logic into another class called ConfigLogger that takes in an instance of Transport, since it makes it a lot cleaner in terms of tests as well.